src/test/java/org/opentripplanner/middleware/controllers/api/OtpUserControllerTest.java

```diff
@@ -136,4 +173,60 @@ void canPreserveSmsConsentDate() throws Exception {
         OtpUser updatedUser = Persistence.otpUsers.getById(otpUser.id);
         Assertions.assertEquals(otpUser.smsConsentDate, updatedUser.smsConsentDate);
     }
+
+    @Test
+    void canAcceptDependentRequest() {
+        String acceptKey = UUID.randomUUID().toString();
+        dependentUserOne.relatedUsers.add(new RelatedUser(
+            relatedUserOne.email,
+            RelatedUser.RelatedUserStatus.PENDING,
+            nickName,
+            acceptKey
+        ));
+        Persistence.otpUsers.replace(dependentUserOne.id, dependentUserOne);
+
+        String path = TrustedCompanion.getAcceptDependentEndPoint(acceptKey);
+        makeGetRequest(path, null);
+
+        relatedUserOne = Persistence.otpUsers.getById(relatedUserOne.id);
+        assertTrue(relatedUserOne.dependents.contains(dependentUserOne.id));
+
+        dependentUserOne = Persistence.otpUsers.getById(dependentUserOne.id);
+        List<RelatedUser> relatedUsers = dependentUserOne.relatedUsers;
+        relatedUsers
+            .stream()
+            .filter(user -> user.email.equals(relatedUserOne.email))
+            .forEach(user -> assertEquals(RelatedUser.RelatedUserStatus.CONFIRMED, user.status));
+    }
+
+    @Test
+    void canInvalidateDependent() {
+        relatedUserTwo.dependents.add(dependentUserTwo.id);
+        Persistence.otpUsers.replace(relatedUserTwo.id, relatedUserTwo);
+        dependentUserTwo.relatedUsers.add(new RelatedUser(
+            relatedUserTwo.email,
+            RelatedUser.RelatedUserStatus.CONFIRMED,
+            nickName
+        ));
+        Persistence.otpUsers.replace(dependentUserTwo.id, dependentUserTwo);
+        relatedUserTwo.delete(false);
+        dependentUserTwo = Persistence.otpUsers.getById(dependentUserTwo.id);
+        RelatedUser relatedUser = dependentUserTwo.relatedUsers.get(0);
+        assertEquals(RelatedUser.RelatedUserStatus.INVALID, relatedUser.status);
+    }
+
+    @Test
+    void canRemoveRelatedUser() {
```

Comment:
Also add a test for a removing a user from the related users list. The related user should then not list that other user as a dependent.

Reply:
See; https://github.com/ibi-group/otp-middleware/pull/256/commits/4a00bfa58c2f747e1f645b4b2b0a4739ac136b6f